Brms 36 matching service

manishsw opened the pull request:
updated with matching service and audit details to store in splunk for firstname, lastname and date of birth.

adamconder committed:
add case class for AuditEvent

adding classes for AuditEvents for:

- England and Wales
- Scotland
- Nortern Ireland

Commit:
BRMS-36-Matching-service: updated for matching service

Commit:
BRMS-40: Update BRMAudit to cover Failed audit event

Commit:
BRMS-36-Matching-service: updated matching service, updated test.

Commit:
BRMS-36-Matching-service: removed unwanted entries.

Commit:
BRMS-36-Matching-service: updated for audit purpose.

Commit:
BRMS-36-Matching-service: Update lastname match to check lastname

Commit:
BRMS-36-Matching-service: Create response package

Commit:
BRMS-36-Matching-service: Insert relevant test objects for MatchingServiceSpec

Commit:
BRMS-36-Matching-service: Code tidy

Commit:
BRMS-36-Matching-service: code tidy...!

Commit:
BRMS-36-Matching-service: Merge branch 'BRMS-40' into BRMS-36-Matching-service

Commit:
BRMS-36-Matching-service: Insert event log in performMatch()

Commit:
BRMS-36-Matching-service: Merge branch 'BRMS-36-Matching-service' of github.com:hmrc/birth-registration-matching into BRMS-36-Matching-service

Commit:
BRMS-36-Matching-service: updated audit keys for splunk data.

Commit:
BRMS-36-Matching-service: added more tests for different case for firstname and lastname

Commit:
BRMS-36-Matching-service: added configuration for matching of firstName, LastName and Date of birth. Updated tests

Commit:
BRMS-36-Matching-service: corrected audit logs.

Commit:
BRMS-36-Matching-service: code cleaning.

chris-ianson commented on app/uk/gov/hmrc/brm/services/MatchingService.scala:
```diff
@@ -0,0 +1,51 @@
+trait MatchingService {
+  val CLASS_NAME: String = this.getClass.getCanonicalName
+
+  def performMatch(input: Payload, response: GroResponse, matchingType: MatchingType.Value): ResultMatch = {
```
Is it worth having a default for matchingType? If no value is passed it defaults to MatchingType.FULL

adamconder replied:
@chris-ianson I would agree with that.

adamconder commented on app/uk/gov/hmrc/brm/services/MatchingAlgorithm.scala:
```diff
@@ -0,0 +1,101 @@
+  }
+
+
+  private def getMatchResult(firstNames : Match, lastNames : Match,  dates: Match) : ResultMatch = {
```
@manishsw I'm going to refactor this method and remove it entirely.

adamconder commented on app/uk/gov/hmrc/brm/services/MatchingService.scala:
```diff
@@ -0,0 +1,51 @@
+    debug(CLASS_NAME, "MatchingType", s"${matchingType}")
+    val algorithm = matchingType match {
+      case MatchingType.FULL => FullMatching
+      case MatchingType.PARTIAL => PartialMatching
```
@manishsw I'm going to add a default case for this which is FullMatching

adamconder committed:
brms-40: refactor methods for partial matching

removing partical matching method and refactoring out the var solution
adding default case for the matching type

adamconder committed:
brms-40 refactor matching audit map
